src/libsyntax/diagnostics/macros.rs

```diff
@@ -1,112 +1,112 @@
 #[macro_export]
 macro_rules! diagnostic_used {
-    ($code:ident) => (
+    ($code:expr) => (
         let _ = $code;
     )
 }
 
 #[macro_export]
 macro_rules! span_fatal {
-    ($session:expr, $span:expr, $code:ident, $($message:tt)*) => ({
+    ($session:expr, $span:expr, $code:expr, $($message:tt)*) => ({
         $crate::diagnostic_used!($code);
         $session.span_fatal_with_code(
             $span,
             &format!($($message)*),
-            $crate::errors::DiagnosticId::Error(stringify!($code).to_owned()),
+            $crate::errors::DiagnosticId::Error($code.to_owned()),
```

Comment:
This seems like an unnecessary change; we can keep the identifiers and continue using `stringify!($code)`. The only thing that's important from a recompilation POV is that `diagnostic_used!(...)` be removed or that if it exists for the purposes of tidy, that it shouldn't do anything (i.e. have an empty expansion).